kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsService.java

```diff
@@ -20,10 +21,26 @@
 @Plugin(configType = Config.class)
 public class VaultKmsService implements KmsService<Config, String, VaultEdek> {
 
+    @SuppressWarnings("java:S3077") // Config is an immutable object
+    private volatile Config config;
```

Comment:
Might be worth documenting why the config field is volatile since its not something we commonly do in Kroxylicious

Reply:
I'll comment it.  The reason for the volatile is that it is possible close will be called by another thread (a shutdown hook).  We want to be sure that  config is safely published.